src/libs/actions/Report.ts

```diff
@@ -2164,7 +2171,18 @@ function leaveRoom(reportID: string, isWorkspaceMemberLeavingWorkspaceRoom = fal
 
     API.write('LeaveRoom', parameters, {optimisticData, successData, failureData});
 
-    if (isWorkspaceMemberLeavingWorkspaceRoom) {
+    const sortedReportsByLastRead = ReportUtils.sortReportsByLastRead(allReports, reportMetadata);
+    // We want to filter out the current report, hidden reports and empty chats
+    const filteredReportsByLastRead = sortedReportsByLastRead.filter(
+        (r) => r?.reportID !== reportID && r?.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN && ReportUtils.shouldReportBeInOptionList(r, '', false, [], {}, true),
```

Comment:
```suggestion
        (report) => report?.reportID !== reportID && report?.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN && ReportUtils.shouldReportBeInOptionList(report, '', false, [], {}, true),
```

Reply:
@luacmartins 
>`report` is already defined in the scope which fails npm run lint and the npm run prettier moves filter conditions to a single line. The formatted code will be like

https://github.com/Expensify/App/pull/31097#discussion_r1457354582

Reply:
Hmm in that case can we name it `currentReport` or something like that?

Reply:
Updated to rename method parameter `reportID` to `currentReportID` and variable `report` to `currentReport`. So that we can use `report` here.